src/runtime/types.cpp

```diff
@@ -1958,6 +1981,17 @@ extern "C" PyObject* PyObject_Init(PyObject* op, PyTypeObject* tp) noexcept {
     // initUserAttrs themselves, though.
     initUserAttrs(op, tp);
 
+    // Initialize the variables declared in __slots__ to NULL.
+    if (tp->tp_flags & Py_TPFLAGS_HEAPTYPE) {
+        BoxedHeapClass* heap_cls = static_cast<BoxedHeapClass*>(tp);
+        if (heap_cls->nslots > 0) {
+            BoxedHeapClass::SlotOffset* slotOffsets = heap_cls->slotOffsets();
+            for (int i = 0; i < heap_cls->nslots; i++) {
+                *(Box**)((char*)op + slotOffsets[i]) = NULL;
+            }
+        }
+    }
```

Comment:
I think we already memset all of the data to zero on allocation -- I changed this from a `... = NULL` to `assert(... == NULL)` and it seems to work ok.